instrumentation/grpctrace/interceptor_test.go

```diff
@@ -428,3 +435,64 @@ func TestServerInterceptorError(t *testing.T) {
 		kv.Int("message.uncompressed_size", 26),
 	}, span.MessageEvents[1].Attributes)
 }
+
+func TestParseFullMethod(t *testing.T) {
+	tests := []struct {
+		fullMethod string
+		name       string
+		attr       []kv.KeyValue
+	}{
+		{
+			fullMethod: "/grpc.test.EchoService/Echo",
+			name:       "grpc.test.EchoService/Echo",
+			attr: []kv.KeyValue{
+				standard.RPCServiceKey.String("grpc.test.EchoService"),
+				standard.RPCMethodKey.String("Echo"),
+			},
+		}, {
+			fullMethod: "/com.example.ExampleRmiService/exampleMethod",
+			name:       "com.example.ExampleRmiService/exampleMethod",
+			attr: []kv.KeyValue{
+				standard.RPCServiceKey.String("com.example.ExampleRmiService"),
+				standard.RPCMethodKey.String("exampleMethod"),
+			},
+		}, {
+			fullMethod: "/MyCalcService.Calculator/Add",
+			name:       "MyCalcService.Calculator/Add",
+			attr: []kv.KeyValue{
+				standard.RPCServiceKey.String("MyCalcService.Calculator"),
+				standard.RPCMethodKey.String("Add"),
+			},
+		}, {
+			fullMethod: "/MyServiceReference.ICalculator/Add",
+			name:       "MyServiceReference.ICalculator/Add",
+			attr: []kv.KeyValue{
+				standard.RPCServiceKey.String("MyServiceReference.ICalculator"),
+				standard.RPCMethodKey.String("Add"),
+			},
+		}, {
+			fullMethod: "/MyServiceWithNoPackage/theMethod",
+			name:       "MyServiceWithNoPackage/theMethod",
+			attr: []kv.KeyValue{
+				standard.RPCServiceKey.String("MyServiceWithNoPackage"),
+				standard.RPCMethodKey.String("theMethod"),
+			},
+		}, {
+			fullMethod: "/pkg.srv",
+			name:       "pkg.srv",
+			attr:       []kv.KeyValue(nil),
+		}, {
+			fullMethod: "/pkg.srv/",
+			name:       "pkg.srv/",
+			attr: []kv.KeyValue{
+				standard.RPCServiceKey.String("pkg.srv"),
+			},
+		},
```

Comment:
I don't know how resilient to unexpected input we need to be (given it looks like `parseFullMethod()` only ever gets called with input that is derived from arguments passed to a gRPC interceptor function), but should we care about what happens if it receives a method that has more than two `/` characters?  e.g. `/pkg/srv/method`? Or one without any `/` characters?  e.g. just `method`?

Feel free to resolve however you think appropriate.

Reply:
I don't have a clear understanding of the unexpected input we need to handle either, the gRPC package is a tough one to understand failure modes of the `FullMethod` field.

I think you're correct that overall we don't need to worry to much about this given [the protobuf spec on field names](https://developers.google.com/protocol-buffers/docs/reference/proto3-spec#identifiers) (though protobuf is only conventionally used as the IDL for gRPC and not a hard requirement I guess). That seems to guarantee the `/something/method` structure.

The more-than-two `/` will result in the method consuming the tail (i.e. `service = "pkg", method = "srv/method"`) and the lack of `/` would mean there wouldn't be a method attribute (just `service = "method"`). Given the malformed nature, I think this behavior is find. I don't know of any specified behavior in the case of these types of failures so I don't know how much testing we should dedicate to this functionality.

Going to leave it as is.